Rt1015 rt1020 sram size fix

jhqian opened the pull request:
sram size for RT1015 and RT1020 needs to be update.

jhqian committed:
dts: arm: nxp: Add dtsi for RT1015 and RT1020

The default flexram configuration for RT1015:
  - ITCM  0x0,         32KB
  - DTCM  0x200000000, 32KB
  - OCRAM 0x20200000,  64KB

The default flexram configuration for RT1020:
  - ITCM  0x0,         64KB
  - DTCM  0x200000000, 64KB
  - OCRAM 0x20200000,  128KB

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
boards: arm: mimxrt1015_evk: update board dts

- update board dts file to refer to nxp_rt1015.dtsi to include sram size
fix.

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
soc: arm: nxp_imx: rt: update flash size of 32KB

- Add 32KB to flash size

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian committed:
boards: arm: mimxrt1020_evk: update board dts

- update board dts file to refer to nxp_rt1020.dtsi to include sram size
fix

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

galak commented on dts/arm/nxp/nxp_rt1015.dtsi:
```diff
@@ -0,0 +1,27 @@
+ * OCRAM: 64KB
+ */
+&itcm0 {
+	compatible = "nxp,imx-itcm";
```
You don't need the compatible here, its already in nxp_rt.dtsi.  Remove from all other nodes as well.

jhqian replied:
thank you, will do.

galak reviewed (dismissed):
You don't need the compatible here, its already in nxp_rt.dtsi. Remove from all other nodes as well.

MaureenHelm reviewed (commented):
Can you check that the ram size is also correct in the board yamls? I think all rt boards except rt1015 use external ram by default, so that may be the only one that needs updating.

jhqian commented:
Yes, @MaureenHelm , only RT1015 is using SRAM, other RTs are using external SDRAM.

jhqian committed:
boards: arm: mimxrt1015_evk: update the ram size

- use ram size with default size of dtcm

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

jhqian commented:
It seems the min_ram defined for the two tests are not correct.

MaureenHelm commented:
> It seems the min_ram defined for the two failed tests are not correct.

Will you please update them?

jhqian commented:
> > It seems the min_ram defined for the two failed tests are not correct.
> 
> Will you please update them?

But I'm not sure the lowest memory it will need, since it depends on the other code for certain device. Or can I just take the size I have on RT1015 and set it to the nearest 4K? for example, if it is 33K-ish on RT1015, I'll set it to 36K?

MaureenHelm commented:
> can I just take the size I have on RT1015 and set it to the nearest 4K? for example, if it is 33K-ish on RT1015, I'll set it to 36K?

Yes

jhqian committed:
tests: change the min_ram for mbedtls and mem_protect

- Change the min_ram to 36K, since they can't be built on RT1015 which
has 32K ram.

Signed-off-by: Ryan QIAN <jianghao.qian@nxp.com>

MaureenHelm commented:
A manual backport is needed for the rt1020 fix; rt1015 was added after 1.14 was released.
